harmoni-web-api/src/main/java/org/harmoniapp/harmoniwebapi/services/AbsenceService.java

```diff
@@ -113,4 +115,38 @@ public AbsenceDto updateAbsence(long id, AbsenceDto absenceDto) {
             throw new RuntimeException("An error occurred: " + e.getMessage(), e);
         }
     }
+
+    /**
+     * Updates the status of an existing Absence.
+     * IMPORTANT it is updated by employer
+     *
+     * @param id the ID of the absence to update
+     * @param absenceDto the AbsenceDto containing the new status of the absence
+     * @return the updated AbsenceDto
+     * @throws IllegalArgumentException if the provided status ID does not exist
+     * @throws RuntimeException if the absence does not exist or if an error occurs during the update process
+     */
+    @Transactional
+    public AbsenceDto updateAbsenceStatus(long id, AbsenceDto absenceDto) {
+        try {
+
+            Absence existingAbsence = repositoryCollector.getAbsences().findById(id)
+                    .orElse(null);
```

Comment:
You can throw an exception at this point, instead of checking later if the value equals null